keras_nlp/samplers/sampler.py

```diff
@@ -45,33 +45,25 @@
 @keras_nlp_export("keras_nlp.samplers.Sampler")
 class Sampler:
     """Base sampler class.
```

Comment:
One more comment. Please add an `Args:` section where you document this argument here.

Reply:
Hey! @mattdangerw 
Regarding this, in the documentation, I added `temperature: float. optional. defaults....`.
Was wondering if in all our other files, we could document parameters which aren't compulsory as `optional` as shown above. Could prove to be more easily interpretable for the users. 🤔

Reply:
So I added the `temperature` parameter to all our current samplers and added unit tests fo them as well. However, for some reason the tests on the `beam sampler` are failing, it would be helpful to know where I am going wrong with it. Thanks!